app/controllers/campaign_changelogs_controller.rb

```diff
@@ -0,0 +1,36 @@
+class CampaignChangelogsController < ApplicationController
+  before_action :authenticate_user!
+  before_action :set_campaign
+  before_action :set_version, only: [:diff, :rollback, :destroy]
+
+  def index
+    @versions = @campaign.versions
+  end
+
+  def create
+    comment = Comment.build_from(@campaign, current_user.id, params[:note])
+    comment.save
+    redirect_to campaign_campaign_notes_path(@campaign)
+  end
+
+  def diff
+    render layout: false
+  end
```

Comment:
⚠️ We should adhere to a strict REST pattern with __all__ controllers/actions. While this may seem pedantic, adhering to this strict policy will pay __big__ dividends in the future.

Consider changing to:

```ruby
# config/routes.rb
resource :diffs, only: [:show]
```

This controller/action now becomes polymorphic in nature and can show diffs on any model with a `paper_trail`.